### PR TITLE
ReporterReader and stochastic tools utilities

### LINKED ISSUE #17391: Include python tools for working with Stocastic Tools JSON output
## Reason

Following #17390 tools for extracting and plotting the statistics JSON output using python

## Design

Create python tools, much like the current PostprocessorValue and VectorPostprocessorValue tools that exist. These need to be able to produce data structures that provide data with time and/or at a specific time step.

## Impact

New features.

### PR DESCRIPTION
**Pull request by zachmprince**
Closes #17391

### CODE REVIEW

File: modules/stochastic_tools/python/visualize_sobol.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    parser.add_argument('-ci', '--confidence-interval', dest='confidence_interval', default=None, nargs=2, type=str, help="Pair of confidence interval levels to use, by default the largest and smallest are used.")
+    parser.add_argument('-t', '--time', default=None, type=float, help="Time to consider, by default the last time is used.")
+    parser.add_argument('-n', '--names', default='{}', type=str, help="Map between the value name and display name in json format. Example: '{\"value1_name\" : \"Value One\", \"value2_name\" : \"Value Two\"}'")
+    parser.add_argument('-p', '--param-names', dest='param_names', nargs='+', default=[], type=str, help="List of names to display for the parameters.")
+    parser.add_argument('-nf', '--number-format', dest='number_format', default='.4g', type=str, help="The number format for tables.")
+    parser.add_argument('-o', '--output', default=None, type=str, help="File name to save figure or table")
+    parser.add_argument('--ignore-ci', dest='include_ci', action='store_false', help="Use this argument to ignore confidence intervals.")
```
Comment (aeslaughter) [inline modules/stochastic_tools/python/visualize_sobol.py:43]: The "dest" doesn't match the argument. However, you don't need "dest" in any of these. `argparse` automatically converts "foo-bar" to "foo_bar"

File: modules/stochastic_tools/python/visualize_sobol.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            ctitle = '$S_i$'
+        elif stat == 'SECOND_ORDER':
+            ctitle = '$S_{i,j}$'
+        colorbar = go.heatmap.ColorBar(title=dict(text=ctitle, side='right'), x=1, y=(i+0.5)/len(vnames), len=1/len(vnames), tickvals=ztick, ticktext=ztick_lab)
+
+        fig.add_trace(go.Heatmap(x=data['param_names'], y=data['param_names'], z=z, colorbar=colorbar), row=i+1, col=1)
+
+    return fig
+
+if __name__ == '__main__':
```
Comment (aeslaughter) [inline modules/stochastic_tools/python/visualize_sobol.py:176]: I would put all this in to a "main()" that returns 0 at the end, then change this to 

```
sys.exit(main())
```

This allows other tools to import the main function desired for wrapping it with other utils

File: modules/stochastic_tools/python/visualize_statistics.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    if xname in data.vector_name.unique():
+        fig = statLine(data, xname)
+    else:
+        fig = statTimeLine(data, xname)
+    if file is None:
+        fig.show()
+    else:
+        plotly.io.write_image(fig, os.path.abspath(file))
+
+if __name__ == '__main__':
```
Comment (aeslaughter) [inline modules/stochastic_tools/python/visualize_statistics.py:248]: same comment on this main

**Review by aeslaughter — CHANGES_REQUESTED** [review]
A few minor changes, the code looks solid. Be careful or you will end up maintaining all the python madness I created.

File: modules/stochastic_tools/python/visualize_sobol.py (hunk 1 above)
Comment (zachmprince) [reply modules/stochastic_tools/python/visualize_sobol.py:43]: I didn't know that! That's cool. The `dest` for `--ignore-ci` would still be different. I can change this if it isn't standard practice?

Comment (aeslaughter) [reply modules/stochastic_tools/python/visualize_sobol.py:43]: I would remove it, doesn't seem like it should be different.

Comment (zachmprince) [reply modules/stochastic_tools/python/visualize_sobol.py:43]: Nevermind, I'll just change it so it matches.

**Review by aeslaughter — DISMISSED** [review]

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**